packages/transactions/tests/builder.test.ts

```diff
@@ -895,6 +896,10 @@ test('Make sponsored STX token transfer', async () => {
   expect(deserializedSponsorSpendingCondition.nonce!.toString()).toBe(sponsorNonce.toString());
   expect(deserializedSponsorSpendingCondition.fee!.toString()).toBe(sponsorFee.toString());
 
+  const spendingCondition = deserializedSponsorSpendingCondition as SingleSigSpendingCondition;
+  const emptySignature = emptyMessageSignature();
+  expect(spendingCondition.signature.data.toString()).not.toBe(emptySignature.data.toString());
+
```

Comment:
Could you make this test a bit more robust? It'd be great if it checked that the contents of the signature are correct.

Reply:
@reedrosenbluth :
We have multiple (4 to 5) test cases considering the sponsored transaction in `builder.test.ts` file

**Question**:
- Do we have to modify all these test cases to verify the signature contents or just the one in this PR.
- Currently the signature content verification is missing in these test cases.

**Proof of concept**:
Below is the sample code snippet to validate the sponsored transaction signature contents. 
```
import {
  AnchorMode,
  makeSTXTokenTransfer,
  sponsorTransaction,
  standardPrincipalCV,
  createSingleSigSpendingCondition,
  publicKeyToString,
  pubKeyfromPrivKey,
  AddressHashMode,
  createStacksPrivateKey,
  TransactionSigner,
  SponsoredAuthorization,
  createSponsoredAuth,
  StacksTransaction,
  TransactionVersion, StandardAuthorization, SingleSigSpendingCondition,
} from '@stacks/transactions';
import { createTokenTransferPayload } from '@stacks/transactions/src/payload';

void (async () => {
  const recipient = standardPrincipalCV('SP3FGQ8Z7JY9BWYZ5WM53E0M9NK7WHJF0691NZ159');
  const amount = 12345;
  const fee = 50;
  const nonce = 2;
  const senderKey = 'edf9aee84d9b7abc145504dde6726c64f369d37ee34ded868fabd876c26570bc01';
  const memo = 'test memo';

  const sponsorKey = '9888d734e6e80a943a6544159e31d6c7e342f695ec867d549c569fa0028892d401';
  const sponsorFee = 123;
  const sponsorNonce = 55;

  const transaction = await makeSTXTokenTransfer({
    recipient,
    amount,
    senderKey,
    fee,
    nonce,
    memo: memo,
    sponsored: true,
    anchorMode: AnchorMode.Any,
  });

  const sponsorOptions = {
    transaction,
    sponsorPrivateKey: sponsorKey,
    fee: sponsorFee,
    sponsorNonce,
  };

  const sponsorSignedTx = await sponsorTransaction(sponsorOptions);

  const sponsoredAuthorized = (sponsorSignedTx.auth as SponsoredAuthorization)
    .sponsorSpendingCondition;
  const sponsoredSpendingCondition = sponsoredAuthorized as SingleSigSpendingCondition;

  const signerAuthorized = (sponsorSignedTx.auth as StandardAuthorization)
    .spendingCondition;

  const signerSpendingCondition = signerAuthorized as SingleSigSpendingCondition;

  // Create same sponsored transaction by steps:
  const payload = createTokenTransferPayload(recipient, amount, memo);
  const addressHashMode = AddressHashMode.SerializeP2PKH;
  const spendingCondition = createSingleSigSpendingCondition(
    addressHashMode,
    publicKeyToString(pubKeyfromPrivKey(senderKey)),
    nonce,
    fee
  );
  const sponsorSpendingCondition = createSingleSigSpendingCondition(
    addressHashMode,
    publicKeyToString(pubKeyfromPrivKey(sponsorKey)),
    sponsorNonce,
    sponsorFee
  );
  const authorization = createSponsoredAuth(spendingCondition, sponsorSpendingCondition);
  const transactionVersion = TransactionVersion.Mainnet;
  const sponsoredTransaction = new StacksTransaction(transactionVersion, authorization, payload);

  const signer = new TransactionSigner(sponsoredTransaction);
  signer.signOrigin(createStacksPrivateKey(senderKey));
  signer.signSponsor(createStacksPrivateKey(sponsorKey));

  const transaction2 = signer.transaction;

  const sponsoredAuthorized2 = (transaction2.auth as SponsoredAuthorization)
    .sponsorSpendingCondition;
  const sponsoredSpendingCondition2 = sponsoredAuthorized2 as SingleSigSpendingCondition;

  const signerAuthorized2 = (transaction2.auth as StandardAuthorization)
    .spendingCondition;

  const signerSpendingCondition2 = signerAuthorized2 as SingleSigSpendingCondition;


  console.log('Sponsored signature data match: ', sponsoredSpendingCondition2.signature.data.toString() === sponsoredSpendingCondition.signature.data.toString());
  console.log('Sponsored signature type match: ', sponsoredSpendingCondition2.signature.type.toString() === sponsoredSpendingCondition.signature.type.toString());

  console.log('Signer signature data match: ', signerSpendingCondition2.signature.data.toString() === signerSpendingCondition.signature.data.toString());
  console.log('Signer signature data match: ', signerSpendingCondition2.signature.type.toString() === signerSpendingCondition.signature.type.toString());

})();

```
@reedrosenbluth  Let me know if the above `code snippet` makes sense in terms of signature content verification or let me know if it's needs any changes or fixes. Based on your thought's i will modify the test cases to validate the signature content's of sponsored transaction. You thorough answer would be appreciated. Thanks.

Reply:
> Could you make this test a bit more robust? It'd be great if it checked that the contents of the signature are correct.

Update: 
• Updated the test case for sponsored transaction
• Added signature contents matching
@reedrosenbluth Give it a review it makes sense or need any further additions.

Reply:
Looks good!